android/src/main/java/com/tailscale/ipn/AppsConfig.java

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
```

Comment:
This file is the primary issue in the PR. It appeared to have been substantially copied from OpenVPN, and carries a GPL license.

There are only so many ways to use the Android APIs in this area and any code to use those APIs will have some resemblances. It needs to be developed afresh from the APIs, and not be GPL code.

Reply:
Sorry for the very late reply @Myned , I am not using tailscale anymore so my motivation for finishing this pull request was gone and on top of that all my Github emails/notifications were in the Spam folder. 

As far as I remember I wrote that class from scratch (minus the method I was pointing in my comment), so perhaps that's the only method that needs to be rewritten but probably @DentonGentry can give you more details since he did the initial review. Remember to also filter the tailscale app from the list of apps as I was pointing in my comment, I don't remember if I did that in the end or not.

Good luck!.

Reply:
Yes, `getInstalledApps()` appears to be the thing to focus on.